rig.js

```diff
@@ -498,87 +496,66 @@ class RigManager {
-      floorHeight,
-      handsEnabled,
-      topEnabled,
-      bottomEnabled,
-      direction,
-      velocity,
-      jumpState,
-      jumpTime,
-      flyState,
-      flyTime,
-      useTime,
-      useAnimation,
-      sitState,
-      sitAnimation,
-      danceState,
-      danceTime,
-      danceAnimation,
-      throwState,
-      throwTime,
-      crouchState,
-      crouchTime,
-    ] = poseArray;
-
-    const peerRig = this.peerRigs.get(peerId);
-
-    if (peerRig) {
-      peerRig.inputs.hmd.position.fromArray(hmdPosition);
-      peerRig.inputs.hmd.quaternion.fromArray(hmdQuaternion);
-
-      peerRig.inputs.leftGamepad.position.fromArray(leftGamepadPosition);
-      peerRig.inputs.leftGamepad.quaternion.fromArray(leftGamepadQuaternion);
-      peerRig.inputs.leftGamepad.pointer = leftGamepadPointer;
-      peerRig.inputs.leftGamepad.grip = leftGamepadGrip;
-
-      peerRig.inputs.rightGamepad.position.fromArray(rightGamepadPosition);
-      peerRig.inputs.rightGamepad.quaternion.fromArray(rightGamepadQuaternion);
-      peerRig.inputs.rightGamepad.pointer = rightGamepadPointer;
-      peerRig.inputs.rightGamepad.grip = rightGamepadGrip;
-
-      peerRig.setFloorHeight(floorHeight);
-      for (let i = 0; i < 2; i++) {
-        peerRig.setHandEnabled(i, handsEnabled[i]);
-      }
-      peerRig.setTopEnabled(topEnabled);
-      peerRig.setBottomEnabled(bottomEnabled);
-      peerRig.direction.fromArray(direction);
-      peerRig.velocity.fromArray(velocity);
-      peerRig.jumpState = jumpState;
-      peerRig.jumpTime = jumpTime;
-      peerRig.flyState = flyState;
-      peerRig.flyTime = flyTime;
-      peerRig.useTime = useTime;
-      peerRig.useAnimation = useAnimation;
-      peerRig.sitState = sitState;
-      peerRig.sitAnimation = sitAnimation;
-      peerRig.danceState = danceState;
-      peerRig.danceTime = danceTime;
-      peerRig.danceAnimation = danceAnimation;
-      peerRig.throwState = throwState;
-      peerRig.throwTime = throwTime;
-      peerRig.crouchState = crouchState;
-      peerRig.crouchTime = crouchTime;
-
-      peerRig.textMesh.position.copy(peerRig.inputs.hmd.position);
-      peerRig.textMesh.position.y += 0.5;
-      peerRig.textMesh.quaternion.copy(peerRig.inputs.hmd.quaternion);
-      localEuler.setFromQuaternion(peerRig.textMesh.quaternion, 'YXZ');
-      localEuler.x = 0;
-      localEuler.y += Math.PI;
-      localEuler.z = 0;
-      peerRig.textMesh.quaternion.setFromEuler(localEuler);
-
-      peerRig.rigCapsule.position.copy(peerRig.inputs.hmd.position);
-    }
+  setPeerAvatarPose(player) {
+    const peerRig = this.peerRigs.get(player.id);
+    if (!peerRig) return;
+
+    const pose = player.pose;
+    const {hmd, leftGamepad, rightGamepad} = peerRig.inputs;
+
+    hmd.position.fromArray(pose.position);
+    hmd.quaternion.fromArray(pose.quaternion);
+
+    if (pose.extra.length === 0) return;
```

Comment:
Why would this ever happen?

Reply:
This was happening early on where pose.extra was blank then later filled. I'll remove and test it.

Reply:
Yeah, it just seems if that were to happen it might be a bug elsewhere, or a race condition/uninitialized state.